### PR TITLE
Flakes support

### PR DESCRIPTION
**Pull request by domenkozar**
~Blocked on https://github.com/cachix/pre-commit-hooks.nix/pull/102~
~Blocked on https://github.com/NixOS/nixpkgs/issues/120752~

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -16,3 +16,21 @@ jobs:
...
+      with:
+          install_url: https://nixos-nix-install-tests.cachix.org/serve/lb41az54kzk6j12p81br4bczary7m145/install
+          install_options: '--tarball-url-prefix https://nixos-nix-install-tests.cachix.org/serve'
+          extra_nix_config: |
+            experimental-features = nix-command flakes
+    - uses: cachix/cachix-action@v10
+      with:
+        name: pre-commit-hooks
+        signingKey: '${{ secrets.CACHIX_SIGNING_KEY }}'
+    - run: nix flake check
```
Comment (Mic92) [inline .github/workflows/ci.yaml:36]: I can also recommend `nix flake show` to validate the flake output schema: https://github.com/Mic92/sops-nix/blob/4f384662a85804fa2bc1bc1f99e70bb468e76f88/.github/workflows/test-flakes.yml#L29

Comment (domenkozar) [reply .github/workflows/ci.yaml]: Thanks!

**Comment by evalexpr** [comment]
@domenkozar is there a documentation update needed here? I have the following snippet:

```
nix-pre-commit-hooks = import sources."pre-commit-hooks.nix";
pre-commit-checks = nix-pre-commit-hooks.run {
    src = ./.;
    hooks = {
      nixpkgs-fmt = {
        enable = true;
        pass_filenames = true;
      };
      nix-linter = {
        enable = true;
        entry = pkgs.lib.mkForce "${pkgs.nix-linter}/bin/nix-linter";
        pass_filenames = true;
        excludes = [ "nix/sources.nix" ];
      };
    };
  };
```

Using `niv` to pin pre-commit-hooks.nix, my sources look like:

```
"pre-commit-hooks.nix": {
        "branch": "master",
        "description": "Seamless integration of https://pre-commit.com git hooks with Nix.",
        "homepage": "",
        "owner": "cachix",
        "repo": "pre-commit-hooks.nix",
        "rev": "c33f92d9d69a73c77061e4fd47234662625cbdf9",
        "sha256": "0hr64z6ypc74fqhqnqpn238bgi2nxzzx8ypc405waapkzlaxx19l",
        "type": "tarball",
        "url": "https://github.com/cachix/pre-commit-hooks.nix/archive/c33f92d9d69a73c77061e4fd47234662625cbdf9.tar.gz",
        "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz"
    }
```

I've been waiting for flakes support so I've been eager to update, just updated and now I get the error:

```
error: attribute 'run' missing

       at /home/wilkins/nixcfg/shell.nix:5:23:

            4|   name = builtins.trace nix-pre-commit-hooks "dev-shell";
            5|   pre-commit-checks = nix-pre-commit-hooks.run {
```

I've checked what the `pre-commit-hooks` import looks like and I see:

```
{ ansible-lint = <CODE>; brittany = <CODE>; cabal-fmt = <CODE>; cargo = <CODE>; clippy = <CODE>; elm-format = <CODE>; fourmolu = <CODE>; hindent = <CODE>; hlint = <CODE>; hpack = <CODE>; hpack-dir = <CODE>; html-tidy = <CODE>; hunspell = <CODE>; nix-linter = <CODE>; nixfmt = <CODE>; nixpkgs-fmt = <CODE>; ormolu = <CODE>; pre-commit = <CODE>; pre-commit-check = <CODE>; prettier = <CODE>; purty = <CODE>; rustfmt = <CODE>; shellcheck = <CODE>; stylish-haskell = <CODE>; terraform-fmt = <CODE>; yamllint = <CODE>; }
```

I'm curious if I'm doing something wrong now? or `run` has been removed.

**Comment by domenkozar** [comment]
You want to use `nix-pre-commit-hooks.lib.run {`

**Comment by evalexpr** [comment]
I'm not seeing `lib` in the trace above at the bottom of my last message, and also get:

> error: attribute 'lib' missing

when I try this, shall I create an issue rather than continuing on this PR? I think we should update the README to mention whatever solution we find as it still mentions `nix-pre-commit-hooks.run` - I can do this after we find a solution.

**Comment by domenkozar** [comment]
I'll provide an example in README :)

**Comment by domenkozar** [comment]
@evalexpr https://github.com/cachix/pre-commit-hooks.nix#nix-flakes-support

**Comment by evalexpr** [comment]
@domenkozar got it working, thanks! I realize my error was that I wasn't importing it as a flake... :facepalm: sorry!

**Pull request merged**